extensions/yaml_support/_parser/lg_units_parser.py

```diff
@@ -0,0 +1,197 @@
+                            connections_dict[key] = value if value != 'OUTPUT' else node.name
+                        
+                        return_info['connections_dict'] = connections_dict
+
+
+                    # checks for publisher & subscriber decorators
+                    publishers:List[str] = []
+                    subscribers:List[str] = []
+                    for decorator in child.decorator_list:
+                        if hasattr(decorator,'func'):
+                            decorator_type:str = self.__construct_type(decorator.func,'')
+                            decorator_type = decorator_type.split('.')[-1]
+
+                            if decorator_type in 'publisher':
+                                arg_type:str = self.__construct_type(decorator.args[0],'')
+                                publishers.append(arg_type)
+
+                            elif decorator_type in 'subscriber':
+                                arg_type:str = self.__construct_type(decorator.args[0],'')
+                                subscribers.append(arg_type)
+                                
+                    
+
+
+
+                    if(hasattr(child.returns,'value')):
+                        type = self.__construct_type(child.returns,type)
+                        return_info['type'] = type
+
+                
+                    class_model.methods[child.name] = {
+                        "args":arguments_info,
+                        "return":return_info,
+                        "publishers":publishers,
+                        "subscribers":subscribers
+                    }
+                    
+        
+            assert isinstance(class_model, LabGraphUnitsModel)
+            self.__classes.append(class_model)
+            self.generic_visit(node)
+
+    
+    def __construct_type(self,value,type):
+        """
+        Recursive method that helps to construct a string 
+        that represents a complex datatype of a class member or a method
+        
+        Args:
+            type: the final string we are trying to construct recursivaly
+            value : an object that contains data about the type of the class member/method
+        """
+        
+        if hasattr(value,'attr'):
+            type = f"{self.__construct_type(value.value,type)}.{value.attr}"
+        
+        elif hasattr(value,'elts'):
+            type = f"{self.__construct_type(value.elts[0],type)}{f'.{type}' if type else ''}"
+
+        elif hasattr(value,'id'):
+            return value.id
+
+        elif isinstance(value,str):
+            return value
+
+        elif hasattr(value,'slice'):
+            type =  f"{self.__construct_type(value.value,type)}.{self.__construct_type(value.slice,type)}{f'.{type}' if type else ''}"                   
+                    
+        
+        elif hasattr(value,'value'):
+            type = f"{self.__construct_type(value.value,type)}{f'.{type}' if type else ''}"
+
+        return type
+        
```

Comment:
Remove redundant spaces.
Please use Flake for formatting: https://github.com/facebookresearch/labgraph/pull/23

Reply:
Done